gridData/mrc.py

```diff
@@ -107,14 +105,24 @@ def read(self, filename):
                                  "alpha={0}, beta={1}, gamma={2}".format(
                                      h.cellb.alpha, h.cellb.beta, h.cellb.gamma))
             # mrc.data[z, y, x] indexed: convert to x,y,z as used in GridDataFormats
-            # mapc, mapr, maps = 1, 2, 3 for Fortran-ordering and 3, 2, 1 for C-ordering. 
-            # Other combinations are possible. We reorder the data for the general case 
-            # by sorting mapc, mapr, maps in descending order.
-            axes_order = np.hstack([mrc.header.mapc, mrc.header.mapr, mrc.header.maps])
-            axes_c_order = np.argsort(axes_order)[::-1]
-            self.array = np.transpose(mrc.data, axes = axes_c_order)
-            self.delta = np.diag([mrc.voxel_size.x, mrc.voxel_size.y, mrc.voxel_size.z])
-            self.origin = np.array([h.origin.x, h.origin.y, h.origin.z])
+            # together with the axes orientation information in mapc/mapr/maps.
+            # mapc, mapr, maps = 1, 2, 3 for Fortran-ordering and 3, 2, 1 for C-ordering.
+            # Other combinations are possible. We reorder the data for the general case
+            # by sorting mapc, mapr, maps in ascending order, i.e., to obtain x,y,z.
+            # mrcfile provides the data in zyx shape (without regard to map*) so we first
+            # transpose it to xyz and then reorient with axes_c_order.
+            #
+            # All other "xyz" quantitities are also reordered.
+            axes_order = np.hstack([h.mapc, h.mapr, h.maps])
+            axes_c_order = np.argsort(axes_order)
+            self.array = np.transpose(np.transpose(mrc.data), axes=axes_c_order)
+            self.delta = np.diag(
+                np.array([mrc.voxel_size.x, mrc.voxel_size.y, mrc.voxel_size.z])[axes_c_order])
+            # the grid is shifted to the CCP4 origin by offset
+            # (assume orthorhombic)
+            offsets = np.hstack([h.nxstart, h.nystart, h.nzstart])[axes_c_order] * np.diag(self.delta)
+            # GridData origin is centre of cell at x=col=0, y=row=0 z=seg=0
+            self.origin = np.hstack([h.origin.x, h.origin.y, h.origin.z])[axes_c_order] + offsets
```

Comment:
The code below works with only one transpose.  Because `mrcfile` transposes the data, we need to reverse the order of `axes_order` before sorting. Note that `voxelsize`  and `origin`  refer to the x, y, and z real space axes and should not be reordered.  However, `nxstart`, `nystart` and `nzstart` should be reordered using `axes_c_order`. I'm not sure why, but I had to do this to match what ChimeraX and VMD are reporting for the example file in #76.
```suggestion
            transpose_order = np.argsort(axes_order[::-1])
            self.array = np.transpose(mrc.data, axes=transpose_order)
            self.delta = np.diag(np.array([mrc.voxel_size.x, mrc.voxel_size.y, mrc.voxel_size.z]))
            # the grid is shifted to the MRC origin by offset
            # (assume orthorhombic)
            offsets = np.hstack([h.nxstart, h.nystart, h.nzstart])[axes_c_order] * np.diag(self.delta)
            # GridData origin is centre of cell at x=col=0, y=row=0 z=seg=0
            self.origin = np.hstack([h.origin.x, h.origin.y, h.origin.z]) + offsets
```

Reply:
Thanks, I am going with your suggestion then. I am not an expert on the MRC format and from the description it wasn't clear to me what corresponds to which direction. 

I didn't include the data from #76 as a new test because it looked as if the included test file 1jzv.ccp4 for CCP4 would be sufficient (it also contains reordered axes). However, if you think that there's something in 1cbs.ccp4 that would make for a good test case then we can include it (especially if 1CBS showed you problems that were not apparent with 1jzv.ccp4).

Reply:
It looks like 1cbs.ccp4 doesn't add anything that isn't in 1jzv.ccp4 already. Both have the same use of `mapc`/`r`/`s`, `nstart` and `origin`. 

Looking back, `nx`, `ny`, `nz`, `nstart` and the data are all in reference to the grid, so the transpose/reorder specified by `mapc`/`r`/`s` must be applied.  OTOH, `mx`, `my`, `mz`, `cella`, `cellb` and `origin` refer to the real-space dimensions and do not require reordering. It would be nice to have additional test cases but I think these properties are handled correctly now.

Reply:
Thanks for double checking, much appreciated!